### PR TITLE
Submit validation on enter keypress

### PR DESCRIPTION
**Pull request by brianseek**
## Description
Restructured form markup and added handler for keydown for shadow dom elements to allow submission on enter key.
Also confirmed that this works as expected with DOB since the Date component is not a web component. 

## Original issue(s)
department-of-veterans-affairs/va.gov-team#40262

## Testing done
Added enter submission to unit test.

## Acceptance criteria
- [ ] Validation form submits on enter key

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/check-in/components/pages/validate/ValidateDisplay.jsx [hunk 1]
```diff
@@ -45,7 +45,15 @@ export default function ValidateDisplay({
     },
     [setDob],
   );
-
+  const handleEnter = e => {
+    if (e.key === 'Enter') {
+      validateHandler();
+    }
+  };
+  const handleFormSubmit = useCallback(e => {
```
Comment (brdunfield) [inline src/applications/check-in/components/pages/validate/ValidateDisplay.jsx:53]: I'm not sure this needs to be a useCallback as it has no dependent state.. I think just a regular function would be fine here (and avoid the lint error)

Comment (brianseek) [reply src/applications/check-in/components/pages/validate/ValidateDisplay.jsx]: you are right! Not sure what my angle there was.

File: src/applications/check-in/components/pages/validate/ValidateDisplay.jsx [hunk 2]
```diff
@@ -103,24 +112,25 @@ export default function ValidateDisplay({
             required
             value={last4Ssn}
             data-testid="last-4-input"
+            onKeyDown={handleEnter}
           />
         )}
+        <button
```
[the remark this answers is not in the document]
Comment (brianseek) [reply src/applications/check-in/components/pages/validate/ValidateDisplay.jsx:118]: Good catch, I had my local at mobile size the whole time. Looks like DS has different style rules for submit buttons. I added a few utility classes to get it sorted out. Also noticed margin was off with it coming inside the form.

**Review by brdunfield — APPROVED** [review]
Looking good! 🦑

**Pull request merged**